### DISCUSSION TITLE
Add HubSpot tracking integration to GitBook space

### DISCUSSION THREAD

**Discussion by cybersyn-christian** [GitbookIO/community#549] [Feature Requests]
I would like to HubSpot tracking to my GitBook space. This requires adding a javascript block like this one to each page on the site:
```
<!-- Start of HubSpot Embed Code -->
<script type="text/javascript" id="hs-script-loader" async defer src="//js.hs-scripts.com/<TRACKING_ID>.js"></script>
<!-- End of HubSpot Embed Code -->
```
This does not seem to be possible with integration permissions, as they don't allow javascript integrations.

Adding an integration that does this would make it possible to integrate HubSpot tracking with our public site. This would make public usage of the site more visible to us and help us understand what parts of the site are used most frequently.

**Comment by addisonschultz** [comment]
Hey @cybersyn-christian! Great request 😄 In fact, our community has [just finished making this integration](https://github.com/GitbookIO/integrations/pull/417)! A big shout out to the Tabnine team.

You can already install it into your own team. Using the following link format: 

https://app.gitbook.com/o/<organization_id>/integrations/hubspot

You can find your organization ID in the URL when working in GitBook, behind the `/o/` identifier in the URL. More info on that here: https://developer.gitbook.com/getting-started/concepts#less-than-organizationid-greater-than

Hope this helps!

**Reply by cybersyn-christian** [reply]
Ah, sorry I didn't see that. Thank you!

**Reply by cybersyn-christian** [reply]
@addisonschultz I think I have enabled the integration in our site and configured it with our tracking ID, but I'm not seeing any response in HubSpot. When I load the GitBook site myself, I'm not seeing the tracker loading.

Is there something I have to do to force the integration to apply on our site?

**Reply by addisonschultz** [reply]
Hmm - @nimast, do you have any pointers? I know you got this working on your end

**Reply by nimrod-codota** [reply]
The integration might take some time to load, that was my experience as well. But then you should see the scripts in the `head` of all pages. Ensure you use the fully qualified URL for the tracking code, including the `\\` and not just the tracking ID, as indicated in the documentation.

**Reply by cybersyn-christian** [reply]
@nimrod-codota That seemed to be the case. It took a few hours before the tracker started to pick up, but now it seems to be working.

Thank you!

**Reply by Jagjeet229** [reply]
Hey @nimrod-codota, I tried adding the Url there like "//js.hs-scripts.com/<TRACKING_ID>.js" on the hubspot integration but its not coming through. Do you know what could be wrong?

**Reply by addisonschultz** [reply]
Hey @Jagjeet229 - You should only need to include the ID of your HubSpot space - not the full URL

**Reply by Jagjeet229** [reply]
Hey @addisonschultz, I'll give that a try, but on the comment above @nimrod-codota said that we need to include the full url?

**Reply by nimast** [reply]
Yes, provide the full URL and give it some time. Also, @addisonschultz, you sent me an email with some changes that need to happen, and I didn't get to them. Might it be the case these are preventing it from running correctly? Perhaps you can post those here so others can contribute them

**Reply by Jagjeet229** [reply]
I tried adding the full url and it did not work for me.

**Reply by nimrod-codota** [reply]
> I tried adding the full url and it did not work for me.

sorry about that. GitBook made some platform changes and asked me to implement them but I'm caught up, please feel free to PR those into my branch or branch out of mine. You will need special permissions to be able to test it though.

**Reply by addisonschultz** [reply]
Hi @Jagjeet229 - thanks for your patience. 

I've updated the integration here, and you can install the new version here: https://app.gitbook.com/integrations/gitbook-hubspot

Thanks @nimast @nimrod-codota for your help on this so far. I've left comments in the original PR if you ever get around to updating it.

**Reply by nimrod-codota** [reply]
tysm @addisonschultz ! I couldn't see any comments on the PR, but if this is now officially supported from GitBook by popular demand that's awesome. I went ahead and removed our integration and installed yours, please let me know if there's anything else required on my end 🙏

**Reply by addisonschultz** [reply]
@Jagjeet229 - do you think you could test this integration by installing it to your organization, and letting me know if it works? 

https://app.gitbook.com/integrations/gitbook-hubspot

**Reply by Jagjeet229** [reply]
Hey @addisonschultz, I'll test and let you know. Thanks

**Comment by danros** [comment]
I have installed the new integration but I am seeing this error:
```
Uncaught ReferenceError: u is not defined
    at HTMLDocument.loadScScript (script.js?version=3.m1unx47:2:5)
```

Problem is here: 

```
function loadScScript() {
    if (!u.getElementById(b)) {
```

**Reply by addisonschultz** [reply]
I've just updated this again - can you let me know if it's working now?

**Comment by Zijian-Zhu** [comment]
Hi, I installed the new intergration to my site and page with inputing only `<TRACKING_ID>` into configuration. But I did not find my tracking id under page source or inspect -> network based on https://knowledge.hubspot.com/reports/how-do-i-know-if-my-hubspot-tracking-code-is-working. 

I do found the my tracking id under source -> v1/intergrations -> gitbook-hubspot. But I saw a red cross for 
```
if (!document.getElementById(b)) {
```
The error is following
```
Uncaught ReferenceError: b is not defined
    at loadScript (script.js?version=4.m2urtev:2:34)
    at script.js?version=4.m2urtev:23:9
    at script.js?version=4.m2urtev:25:3
```
I am not sure hubspot intergration is working or not. Please take a look thx

**Reply by addisonschultz** [reply]
Just made another update - can you try it again for me? 😄

**Reply by Zijian-Zhu** [reply]
Thank you, I find my tracking id now

**Comment by ewgenius** [comment]
Hello @addisonschultz, thanks for the integration! I've installed https://app.gitbook.com/integrations/gitbook-hubspot on our website and it works, but there is an issue with gitbook dark mode - hubspot chat iframe has white background.

This can be solved with css, but there is no official way to add custom code to gitbook.
https://community.hubspot.com/t5/Tickets-Conversations/Unwanted-background-behind-chat-widget/td-p/667587

```css
#hubspot-messages-iframe-container {
  color-scheme: light;
}
```

Is there any way to add this into the integration?

![CleanShot 2025-03-18 at 16 15 22@2x](https://github.com/user-attachments/assets/7bee479d-14a1-49cd-84f5-0040f4865f78)
![CleanShot 2025-03-18 at 16 16 49@2x](https://github.com/user-attachments/assets/782b3190-5720-44a9-8d3d-0a37eca47580)

**Comment by yangxiling** [comment]
I installed the integration, but I entered the content wrongly the first time. I filled in all the content of src, //js.hs-scripts.com/<TRACKING_ID>.js.

My page reported a 404 error when loading the hubspot tracking code. Request address:
https://js.hs-scripts.com/%3Cscript%20type=%22text/javascript%22%20id=%22hs-script-loader%22%20async%20defer%20src=%22// js.hs-scripts.com/22114441.js%22%3E%3C/script%3E.js

I modified the content in the integration to only have TRACKING_ID, but it has not been updated when the page is loading.

May I ask what might be the reason for this
